dask_expr/frameexpr.py

```diff
@@ -0,0 +1,1375 @@
+from __future__ import annotations
+
+import functools
+import numbers
+import operator
+from collections import defaultdict
+from collections.abc import Mapping
+
+import dask
+import pandas as pd
+from dask.base import normalize_token, tokenize
+from dask.dataframe import methods
+from dask.dataframe.core import (
+    _get_divisions_map_partitions,
+    _get_meta_map_partitions,
+    apply_and_enforce,
+    is_dataframe_like,
+    is_index_like,
+    is_series_like,
+    make_meta,
+)
+from dask.dataframe.dispatch import meta_nonempty
+from dask.utils import M, apply, funcname
+
+from dask_expr.expr import Expr
+
+replacement_rules = []
+
+no_default = "__no_default__"
+
+
+class FrameExpr(Expr):
```

Comment:
Maybe?

```python
class Frame(Expr)
```

Or is that weird for scalars?  (maybe scalars shouldn't be Frames?)

In general I dislike.  

```python
class FooBar(Bar)
```

It always feels needlessly wordy to me.  This is subjective though and I'm happy to be overruled.

Reply:
Renamed